File: python/205_isomorphic_strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        dict_s = {}
        dict_t = {}

        map_s = []
        map_t = []

        i = 0
        while i < len(s):
            if s[i] in dict_s:
                map_s.append(dict_s[s[i]])
            else:
                dict_s[s[i]] = i
                map_s.append(dict_s[s[i]])

            if t[i] in dict_t:
                map_t.append(dict_t[t[i]])
            else:
                dict_t[t[i]] = i
                map_t.append(dict_t[t[i]])
            i += 1
        return map_s == map_t

File: cpp/205.isomorphic_strings.cpp
#include <string>
#include <unordered_map>
#include <vector>

class Solution {
 public:
    bool isIsomorphic(std::string s, std::string t) {
        std::unordered_map<char, int> mapS;
        std::unordered_map<char, int> mapT;

        std::vector<int> listS;
        std::vector<int> listT;

        for (int i = 0; i < s.size(); ++i) {
            if (mapS.contains(s[i])) {
                listS.push_back(mapS[s[i]]);
            } else {
                mapS[s[i]] = i;
                listS.push_back(mapS[s[i]]);
            }

            if (mapT.contains(t[i])) {
                listT.push_back(mapT[t[i]]);
            } else {
                mapT[t[i]] = i;
                listT.push_back(mapT[t[i]]);
            }
        }

        for (int i = 0; i < listS.size(); ++i) {
            if (listS[i] != listT[i]) {
                return false;
            }
        }

        return true;
    }
};
